File: Python/2706-Buy Two Chocolates.py
class Solution:
    def buyChoco(self, prices: list[int], money: int) -> int:
        a = 0
        b = 1
        for i in range(2, len(prices)):
            if prices[b] >= prices[i]:
                if prices[a] > prices[b]:
                    a = b
                b = i
            elif b != i and prices[a] > prices[i]:
                a = i

        if prices[a] + prices[b] <= money:
            return money - (prices[b] + prices[a])
        return money

File: C++/2706-Buy Two Chocolates.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int buyChoco(vector<int>& prices, int money)
    {
        int a = 0, b = 1;
        for (int i = 2; i < prices.size(); ++i)
        {
            if (prices[b] >= prices[i])
            {
                if (prices[a] > prices[b])
                    a = b;

                b = i;
            }
            else if (b != i && prices[a] > prices[i])
                a = i;

        }

        if (prices[a] + prices[b] <= money)
            return money - (prices[b] + prices[a]);
        return money;
    }
};
